config/Dockerfiles/tests.d/azure/02_azure_virtual_network

```diff
@@ -8,14 +8,17 @@
 DISTRO=${1}
 PROVIDER=${2}
 TARGET="azure_vn"
-pushd docs/source/examples/workspaces/${PROVIDER}
+pushd "docs/source/examples/workspaces/${PROVIDER}"
 function clean_up {
     set +e
-    linchpin -w . -vvv --template-data '{ "distro": '${DISTRO}""'}' destroy "${TARGET}"
+    linchpin -w . -vvv --template-data '{ "distro": "'"${DISTRO}"'", "profile": "key"}' destroy "${TARGET}"
     rm -rf resources/ inventories/
 }
 
 trap clean_up EXIT
 
 
-linchpin -w . -vvv --template-data '{ "distro": '${DISTRO}""'}' up "${TARGET}"
+linchpin -w . -vvv --template-data '{ "distro": "'"${DISTRO}"'", "profile": "key"}' up "${TARGET}"
+clean_up
+
+linchpin -w . -vvv --template-data '{ "distro": "'"${DISTRO}"'", "profile": "ad"}' up "${TARGET}"
```

Comment:
Do you need to destroy the `ad` profile as well as the `key` profile?

Reply:
The 'profile' variable is for selecting the credential for testing, active directory 'ad' or service principle 'key'.

Reply:
It looks like you're still creating two resources and destroying one

Reply:
the "trap" is triggered when the script ends or fails so that's why there is only one cleanup call.